store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package mongo
+
+import (
+	"context"
+
+	"github.com/mendersoftware/go-lib-micro/log"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	mstore_v1 "github.com/mendersoftware/go-lib-micro/store"
+	mstore "github.com/mendersoftware/go-lib-micro/store/v2"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/mendersoftware/deviceauth/model"
+)
+
+const (
+	findBatchSize = 100
+)
+
+type migration_2_0_0 struct {
+	ds  *DataStoreMongo
+	ctx context.Context
+}
+
+// Up creates an index on status and id in the devices collection
+func (m *migration_2_0_0) Up(from migrate.Version) error {
+	// _false := false
+	// _true := true
+	// var expireAfterSeconds int32 = 0
+	// tokenExpirationIndexName := "TokenExpiration"
+	currentDbName := mstore_v1.DbFromContext(m.ctx, DbName)
+	logger := log.FromContext(m.ctx)
+
+	collectionsIndexes := map[string]struct {
+		Indexes []mongo.IndexModel
+	}{
+		DbDevicesColl: {
+			Indexes: []mongo.IndexModel{
+				// create device index on status and id
+				{
+					Keys: bson.D{
+						{Key: model.TenantIDField, Value: 1},
+						{Key: model.DevKeyStatus, Value: 1},
+						{Key: model.DevKeyId, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(indexDevices_Status).
+						SetUnique(false),
+				}, {
+					Keys: bson.D{
+						{Key: model.DevKeyIdDataSha256, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(indexDevices_IdentityDataSha256).
+						SetUnique(true),
+				},
+			},
+		},
+		DbAuthSetColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{
+						{Key: model.TenantIDField, Value: 1},
+						{Key: model.AuthSetKeyIdDataSha256, Value: 1},
+						{Key: model.AuthSetKeyPubKey, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(indexAuthSet_IdentityDataSha256_PubKey).
+						SetUnique(true),
+				},
+			},
+		},
```

Comment:
Reviewing the queries we make to the `auth_sets` collection, we'll also need an index key on the `device_id` field too. That is,
```json
{"tenant_id": 1, "device_id": 1}
```
You may also consider appending the a key on the timestamp as well to get the ability to sort the autsets for a device by date.